### PR TITLE
Add metrics for submission and time in queue

### PR DESCRIPTION
**Pull request by edwinalu**
Add the following new metrics:
- submit flow success, fail and skip
- queue wait time (time between when a flow is submitted, to when an executor starts executing)
- flow setup time (time to setup a flow, before executing).

The time that a flow spends in PREPARING state is queue wait time + flow setup time. These metrics will help give more insight into how much time is spent in preparing state, and in which phases.
    
Flow submission is when a user requests a flow to be executed, or when a flow is scheduled to run.  Flow submission will add the flow to the queue. Flow dispatch is when the flow is assigned to an executor; currently this time also includes the time to setup the flow.

### CODE REVIEW

**Comment by burgerkingeater** [comment]
would be great to call out that the time flow spends in preparing state is queue wait time + flow setup time.

File: azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java [hunk 1]
```diff
@@ -28,6 +30,18 @@
  */
 @Singleton
 public class CommonMetrics {
+  public static final String FLOW_FAIL_METER_NAME = "flow-fail-meter";
```
Comment (burgerkingeater) [inline azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java]: thanks for refactoring.

**Comment by burgerkingeater** [comment]
you might need to resolve the conflicts first.

**Comment by edwinalu** [comment]
Add the following new metrics:

    * submit flow success, fail and skip
    * queue wait time (time between when a flow is submitted, to when an executor starts executing)
    * flow setup time (time to setup a flow, before executing).

The time that a flow spends in PREPARING state is queue wait time + flow setup time. These metrics will help give more insight into how much time is spent in preparing state, and in which phases.

Flow submission is when a user requests a flow to be executed, or when a flow is scheduled to run. Flow submission will add the flow to the queue. Flow dispatch is when the flow is assigned to an executor; currently this time also includes the time to setup the flow.

File: azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java [hunk 2]
```diff
@@ -28,6 +30,18 @@
...
+  public static final String DISPATCH_FAIL_METER_NAME = "dispatch-fail-meter";
+  public static final String DISPATCH_SUCCESS_METER_NAME = "dispatch-success-meter";
+  public static final String SEND_EMAIL_FAIL_METER_NAME = "send-email-fail-meter";
+  public static final String SEND_EMAIL_SUCCESS_METER_NAME = "send-email-success-meter";
+  public static final String SUBMIT_FLOW_SUCCESS_METER_NAME = "submit-flow-success-meter";
+  public static final String SUBMIT_FLOW_FAIL_METER_NAME = "submit-flow-fail-meter";
+  public static final String SUBMIT_FLOW_SKIP_METER_NAME = "submit-flow-skip-meter";
+  public static final String OOM_WAITING_JOB_COUNT_NAME = "OOM-waiting-job-count";
+  public static final String QUEUE_WAIT_HISTOGRAM_NAME = "queue-wait-histogram";
+  public static final String FLOW_SETUP_TIMER_NAME = "flow-setup-timer";
```
Comment (burgerkingeater) [inline azkaban-common/src/main/java/azkaban/metrics/CommonMetrics.java]: should flow-setup-timer be in the ExecMetrics?

**Pull request merged**